features/lwipstack/LWIPStack.cpp

```diff
@@ -197,6 +201,12 @@ void LWIP::tcpip_init_irq(void *eh)
 /* LWIP network stack implementation */
 LWIP::LWIP()
 {
+    if (instance) {
```

Comment:
I don't understand why you're doing this - allowing construction other than by `get_instance`, but if you are, you need to fix `get_instance` itself so that if someone has already constructed an `LWIP` and filled in `_instance`, it returns that, and doesn't create another one.

Then you wouldn't be breaking `socket_callback` above, or any other piece of code that uses `OnboardNetworkStack::get_default_instance()`.

Reply:
Oh, is this really about avoiding a potential mutex lock in some path in `socket_callback` due to the `get_instance`? Commit message said it was about heap allocation.

Reply:
No, the check was just to prevent someone from creating the stack e.g. by new and later by get default instance. Removed the check since do not need to allow construct other than get instance.

Reply:
The constructor is (was?) private to force use of `get_instance`, because of the inherent singleton nature. But I would be open to supporting manual construction to permit a different allocation strategy, as long as `get_instance` still works after that has been done.

Reply:
Yes, that could be future requirement.